client/state/index.js

```diff
@@ -0,0 +1,27 @@
+/**
+ * External dependencies
+ */
+import thunkMiddleware from 'redux-thunk';
+import { createStore, applyMiddleware, combineReducers } from 'redux';
+
+/**
+ * Internal dependencies
+ */
+import sharing from './sharing/reducer';
+import sites from './sites/reducer';
+import ui from './ui/reducer';
+
+/**
+ * Module variables
+ */
+const reducer = combineReducers( {
+	sharing,
+	sites,
+	ui
+} );
+
+export function createReduxStore() {
```

Comment:
>I will change this in #1393 to export not the function, but singleton - already instantiated store.

I disagree with this move. It's important that stores are instantiated on the fly. This goes back to server-side rendering concerns about having a singleton instance being reused between requests. On the server-side, we'll need to create a new store for every request. If you need access to the store from elsewhere in the app, you should be able to pass it along from the route context value, without the need for pulling it in via module import.